idp_task1.cpp

```diff
@@ -1,55 +1,61 @@
 #include "socket_wrapper.hpp"
-#include "util.h"
+#include "socket_exception.hpp"
+#include <iostream>
+#include <string.h>
+#include <arpa/inet.h>
 
 static const int LISTENQ = 16;
 static const int DEFAULT_ERROR = -1;
 static const int SERVER_PORT = 8877;
-static const int ECHO_STR_LEN = 6;                                            
+static const char* ECHO_STR_PREFIX = "echo: ";
+static const int ECHO_STR_LEN = strlen(ECHO_STR_PREFIX);
 
 void createTestTCPServer()
 {
     try 
     {
-    std::unique_ptr<SocketWrapper> server_socket = std::make_unique<SocketWrapper>(AF_INET, SOCK_STREAM, 0, SERVER_PORT);
+        Socket_sptr server_socket = std::make_shared<Socket>(AF_INET, SOCK_STREAM, 0);
 
-    server_socket->Bind();
+        server_socket->Bind(INADDR_ANY, htons(SERVER_PORT));
 
-    server_socket->Listen(LISTENQ);
+        server_socket->Listen(LISTENQ);
 
-    while (true) 
-    {
-		std::unique_ptr<SocketWrapper> client_socket = server_socket->Accept();
+        while (true) 
+        {
+            Socket_sptr client_socket = server_socket->Accept();
 
-		int n = 0;
-		int maxlen = 100;
-		char buffer[maxlen];
+            int received_data_length = 0;
+            int maxlen = 100;
```

Comment:
As this value is constant, maybe we can place it with other constants in global scope?
We'll get rid of them once.